gprMax/waveforms.py

```diff
@@ -102,6 +103,12 @@ def calculate_value(self, time, dt):
                 ampvalue = 0
             else:
                 ampvalue = self.uservalues[index]
+        elif self.type == 'userfunc':
+                            
+            if self.userfunc is None:
+                ampvalue = 0
+            else:
+                ampvalue = float(self.userfunc(time))
```

Comment:
Does this not rely on an exact match between the absolute time passed into this function, and the time vector given in the excitation file? Which would mean matching two floating point numbers?